### ISSUE TITLE
Markdown header IDs differ from Marked and Jupyter Notebook

### ISSUE THREAD

**Issue by agoose77** [jupyterlab/jupyterlab#5877]
The generated IDs from Markdown headers differ from those generated by Marked, and hence from those generated in Jupyter Notebook.

Take this cell, for example:
```markdown
## *Header*
```

The generated HTML in Jupyter Lab is:

```html
<h2 id="%3Cem%3EHeader%3C%2Fem%3E"><em>Header</em></h2>
```

Whilst the HTML output in Jupyter Notebook is:
```html
<h2 id="Header"><em>Header</em><a class="anchor-link" href="#Header">¶</a></h2>
```

The latter is desirable, as formatting shouldn't affect the textual representation of the ID, in my opinion.

In my lack of experience, I wonder whether we can just use the Marked implementation directly, and append the relevant Jupyter css tags. See #5876

**Comment by ian-r-rose** [comment]
Thanks for the report @agoose77, I believe this was fixed in #5772 (thanks @gavinhenderson!).

The fix will be in the next release. Closing this as resolved, but please comment here if you think that is not the case.

**Comment by ian-r-rose** [comment]
Also, I think you are right that we should consider using the marked implementation directly. We will probably revisit that (and possibly using a different markdown renderer altogether) after 1.0.

cf #5770
